ludwig/backend/ray.py

```diff
@@ -571,24 +574,29 @@ def to_tensors(df: pd.DataFrame) -> pd.DataFrame:
                 df[c] = df[c].astype(TensorDtype())
             return df
 
-        trainer_kwargs = {**get_trainer_kwargs(), **self.trainer_kwargs}
+        # TODO(shreya): self.trainer_kwargs should have the correct resources; debug.
+        # trainer_kwargs = {**get_trainer_kwargs(), **self.trainer_kwargs}
+        trainer_kwargs = {**self.trainer_kwargs, **get_trainer_kwargs()}
         resources_per_worker = trainer_kwargs.get("resources_per_worker", {})
-        num_cpus = resources_per_worker.get("CPU", 1)
         num_gpus = resources_per_worker.get("GPU", 0)
-
-        dask_dataset = (
-            dataset.ds.map_batches(to_tensors, batch_format="pandas")
-            .map_batches(
-                batch_predictor,
-                batch_size=self.batch_size,
-                compute="actors",
-                batch_format="pandas",
-                num_cpus=num_cpus,
-                num_gpus=num_gpus,
-            )
-            .to_dask()
+        num_cpus = resources_per_worker.get("CPU", (1 if num_gpus == 0 else 0))
+
+        dask_dataset = dataset.ds.map_batches(
+            to_tensors,
+            batch_format="pandas",
+            num_cpus=num_cpus,
+            num_gpus=num_gpus,
```

Comment:
The `to_tensors` conversion shouldn't need to be done on GPU. Can we remove the `num_cpus` and `num_gpus` requests here?